### ISSUE TITLE
[Issue]: Cannot launch on Debian bookworm

### ISSUE THREAD

**Issue by sevenrats** [jellyflix-app/jellyflix#46]
### Please describe your bug

I download the executable and install the noted dependencies. There are undocumented dependences but thats out of scope for this issue report.  I download all the dependencies and attempt to run the downloaded binary.  It fails because a library libmpv.1.so does not exist.  My system (bookworm) only has access to libmpv2.

### Jellyflix Version

0.9.13

### Jellyfin Version

10.8.13

### Environment

```markdown
- OS:Debian
- OS Version: Bookworm
- Playback Method: Didnt make it that far
```

**Comment by sevenrats** [comment]
this issue can be worked around with this symlink after installing libmpv2:
```
sudo ln -s /usr/lib/x86_64-linux-gnu/libmpv.so.2 /usr/lib/x86_64-linux-gnu/libmpv.so.1
```

**Comment by jdk-21** [comment]
You can add this workaround to your PR #54, if you want.

**Comment by sevenrats** [comment]
closing as a dupe of #48 then

**Comment by sevenrats** [comment]
reopening in case there is some way we can implement a workaround in code without needing root perms and without needing the user to make a symlink in order to run a program.  maybe there is some way we can pass an option to mpv or something to make it look somewhere else,

**Comment by jdk-21** [comment]
The best way to solve this issue would be to contribute to the package (media-kit) that we use for playback. As it turns out they already talked about this issue: https://github.com/media-kit/media-kit/issues/507
And the best way to prevent it is packaging Jellyflix.

We can use this issue to track progress for a packaged version like flatpak

**Comment by BLumia** [comment]
Just leave a note that this issue also happens on Arch Linux.

I personally suggest distributing AppImage package and maybe also submit this app to FlatHub.

**Comment by sevenrats** [comment]
I would rather document the symlink forever than go anywhere near any linux packaging solution. In eras like this one where we are straddling Debuntu generations, we should simply have the CI run on each.  Packaging sucks and causes disprortionate explosion of the support burden.  If the community wants a flatpak version, let the community support a flatpak version.

**Comment by sevenrats** [comment]
seriously flatpak makes it literally impossible to triage jellyfin media player. please don't do it.

**Comment by BLumia** [comment]
> seriously flatpak makes it literally impossible to triage jellyfin media player.

Since one can always `flatpak run --command=sh the.app.id` for debugging or other purpose, could you explain in details why do you get such conclusion?

**Comment by sevenrats** [comment]
Because it massively increases the support surface. It's not about whether or not you can debug a flatpak build, its about whether or not we have the time to wade through twice as many issues.  Woops I meant three times because then somebody needs AppImages.  Woops I meant 4 times because then somebody needs snaps.
If it is so easy to package something for flatpak and triage the issues related to it then somebody should just take responsibility for it downstream. How hard could it be?

**Comment by BLumia** [comment]
> its about whether or not we have the time to wade through twice as many issues. Woops I meant three times because then somebody needs AppImages. Woops I meant 4 times because then somebody needs snaps.

Yeah, it's your right to have the final word about which platform you plan to officially support, but what I want to say is, the current method to ship binaries is not even a correct way to ship binaries **since you rely on what's included in your CI environment**.

The reason why AppImages is suggested is because it's the closest solution to package a self-contained package. There is NO container or sandbox involved. Very similar to vendor the required `.so` files and ensure your program load the file that you vendored but it's a portable single-file executible. Flatpak is more complex than AppImage but it's a cross-distro solution so if you like, you can even only treat Flatpak support as the only supported platform.

> we should simply have the CI run on each. 

You know how many debian-based distribution are there, and in case you forget, there are so many rolling release distros (like the previously mentioned Arch Linux). Isn't supporting each of them harder than supporting only a cross-distro solution? And how it's even gonna work if the distro is a rolling release distro if you choose to rely on the shared libraries installed on the system?

To be clear, I didn't mean you should offer official support for all existing packaging solutions, but if you really want to save time on supporting each distro, supporting Flatpak for example, is a way more cleaner way than having different CI to build package for different distros.

Still, it's your right to have the final word.

**Comment by sevenrats** [comment]
Well, firstly let me clarify that it's not my right to have the final word, it's Jonathan's.  These are just my strongly held opinions based on my experience triaging and supporting jellyfin clients.
Second, let me clarify that I dont suggest we support every debian distro. I would just support Debuntu Stable and Debuntu Next.
Everything else should be downstream because as soon as Jonathon's honeymoon with this software is over all these moving pieces you want to add are going to turn to rust.
Its flutter, guys.  If you want an Arch build, build it,its so easy its crazy. And then it will be YOUR responsibility to triage and respond to all the issues about "HDR doesn't work when my cat walks by, only effects flatpak."

**Comment by sevenrats** [comment]
Here, check it out: https://github.com/jellyfin/jellyfin-media-player/issues?q=is%3Aissue+is%3Aopen+flatpak
Fully 20% of our open tickets.  At this point everybody involved with the project wants to rip it out but we don't want the users to pitchfork us.  Jellyflix should have the wisdom to keep itself out of the same predicament.

**Comment by BLumia** [comment]
> Well, firstly let me clarify that it's not my right to have the final word, it's Jonathan's. These are just my strongly held opinions based on my experience triaging and supporting jellyfin clients.

Well, then we both made our point clear already.

> Its flutter, guys. If you want an Arch build, build it,its so easy its crazy.

I mentioned Arch as an example. Arch is NOT the only rolling-release distro that exists. I DO compile it and use it by myself btw (as I also mentioned before).

> I would just support Debuntu Stable and Debuntu Next.

I assume you mean Debian/Ubuntu stable and next, the how does the "next" part work? Debian `sid` For example, can also be considered as a rolling release repo. Or if you mean Debian `testing`, then before its repo freeze starts, it can also be considered as a rolling release repo. In such cases you will still have exactly the same issue that this issue thread is for.

> Here, check it out: https://github.com/jellyfin/jellyfin-media-player/issues?q=is%3Aissue+is%3Aopen+flatpak
Fully 20% of our open tickets. At this point everybody involved with the project wants to rip it out but we don't want the users to pitchfork us.

This should mean you guys should take the Flatpak release more seriously! If Flatpak is officially unsupported from your end (since you say "everybody involved with the project wants to rip it out"), you should remove it.
